src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker_Patterns.cs

```diff
@@ -168,10 +169,11 @@ public override BoundNode VisitITuplePattern(BoundITuplePattern node)
                     }
                     break;
                 case BoundNegatedPattern p:
-                    // PROTOTYPE(ngafter): what is the right behavior here?
+                    LearnFromAnyNullPatterns(inputSlot, inputType, p.Negated);
```

Comment:
>LearnFromAnyNullPatterns(inputSlot, inputType, p.Negated); [](start = 20, length = 58)

This seems strange. Seems like we learn the same thing from `x is not null` as from `x is null`. I'm not seeing the negation.

Reply:
Either one is a pure null test, and lets us infer that the input may have been null.

---
In reply to: [381009686](https://github.com/dotnet/roslyn/pull/41503#discussion_r381009686) [](ancestors = 381009686)